drivers/modem/gsm_ppp.c

```diff
@@ -1160,6 +1178,47 @@ void gsm_ppp_start(const struct device *dev)
 #endif
 }
 
+static void gsm_soft_reset(struct gsm_modem *gsm)
+{
+	if (gsm->soft_reset_requested) {
+		LOG_INF("Soft reseting modem");
+		int retry = 10;
+		while (retry > 0)
+		{
+			retry--;
+			int ret = modem_cmd_send_nolock(&gsm->context.iface,
+									&gsm->context.cmd_handler,
+									&response_cmds[0],
+				    			ARRAY_SIZE(response_cmds),
+									"AT+CFUN=0",
+									&gsm->sem_response, K_SECONDS(2));
+			if(ret == 0) {
+				break;
+			}
```

Comment:
we'll fail silently here. Should we add a print or not worth it?

Reply:
In some network loss events modem goes to some unknown states and won't answer to AT commands and trying more is just a safety guard but if doesn't then there is nothing we can do other than power cycling modem. I think it's good enough. But I can add an error log here